src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -324,7 +325,12 @@ object GetNewSelectedItem(SelectionChangedEventArgs args)
 
 			if (SelectedItem != newSelectedItem)
 			{
+				//This Yield is necessary here, to allow selection of items that have children.
+				//Preventing errors in UIElement PointerReleasedEvent
+				//The given key 'UITests.Shared.Microsoft_UI_Xaml_Controls.TreeViewTests.TreeViewItemSource' was not present in the dictionary.
+				await Task.Yield();
 				SelectedItem = newSelectedItem;
+
```

Comment:
Async void cannot be used to fix issues particularly when handling this type of changes as it might introduce very hard to reproduce race conditions. Let's find another solution.

@dr1rrb does this look familiar? @MartinZikmund is there something similar that happens in the original TreeView?